### PR TITLE
--WITHDRAWN--

### LINKED ISSUE #12958: [Bug][move-compiler-v2] running V2 tests in aptos-move/move-examples runs unit tests for stdlib 8 times
# 🐛 Bug

It looks like the V2 unit test framework is keeping tests for dependences as well as the target files.  This is apparently visible by running
```
MOVE_COMPILER_V2=true cargo test 
```
in aptos-move/move-examples/.

I'll have to sort out why that is happening.

### PR DESCRIPTION
**Pull request by brmataptos**
## Title

Fix #12958: tests being run from library code when aptos-move/move-examples are run with `MOVE_COMPILER_V2=true cargo test`.

## Description

Fixes #12958.

Change compiler-v2 sources and dependencies from `Vec<String>`
filenames to `Vec<PackagePaths>` full package descriptors.
This attaches a package name to input files, allowing a
filter to limit unit tests to the current package.  Since
these are bigger structures, we pass `&` instead of object.

There is some messiness due to `PackagePaths` really being
a generic type with default instantiation using
`move_symbol_pool::Symbol` which isn't common in the
move-model code.

Also, it turns out that disassembler uses script module
names whose name mangling has changed, so there is a
change to `move-cli/src/base/disassemble.rs`, along with
a more verbose error (listing the available functions)
in the case of a failure to find the requested module,
which was useful debugging and was left in.

Since related code was touched in EVM-related code in
`third_party/move/tools/move-unit-test/src/test_runner.rs`
I fixed it to compile but there is still an issue with
the `named_address_map` which I haven't debugged.
(See Issue #13119.)

## Key Areas to Review

This can be refactored later, so keep that in mind.  It's still a bit
messy, but if you can understand what's going on (after reading the
description above) it may be easier to wait until we separate
compilers to do some of the remaining cleanup.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
Cargo test run with various options (`MOVE_COMPILER_V2=true` and not).

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/tools/move-package/src/compilation/compiled_package.rs [hunk 1]
```diff
@@ -748,21 +745,19 @@ impl CompiledPackage {
...
-                resolution_graph.build_options.generate_docs,
-                resolution_graph.build_options.generate_abis,
-                optional_global_env,
-            ) {
-                (false, false, Some(env)) => env, // Use V2 generated model if not for docgen or abigen
-                _ => run_model_builder_with_options_and_compilation_flags(
-                    // Otherwise, use V1 generated model
-                    vec![sources_package_paths],
-                    deps_package_paths.into_iter().map(|(p, _)| p).collect_vec(),
+            let model = if let Some(env) = optional_global_env {
```
Comment (rahxephon89) [inline third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: With this change, if you run e2e move test with V2 flag, the doc will be different from what V1 generates.

File: third_party/move/move-compiler-v2/src/options.rs [hunk 2]
```diff
@@ -19,13 +19,10 @@ use std::{
 #[derive(Parser, Clone, Debug)]
 #[clap(author, version, about)]
 pub struct Options {
+    #[clap(skip)]
```
Comment (rahxephon89) [inline third_party/move/move-compiler-v2/src/options.rs:22]: Do we expose the compiler option to the user? cc @wrwg

Comment (brmataptos) [reply third_party/move/move-compiler-v2/src/options.rs:22]: I don't see a pathway for that.

File: third_party/move/tools/move-package/src/compilation/compiled_package.rs (hunk 1 above)
Comment (brmataptos) [reply third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: I don't see it.  Can you please tell me a command line to show this?

To be honest, I'm surprised this works.  I had to undo a similar thing in the previous PR to get it to land.  But I guess now things are somehow happier.  We may be really running V1 under the covers here; I haven't tried enabling `MOVE_COMPILER_BLOCK_V1` recently.

Comment (rahxephon89) [reply third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: You could try to run `MOVE_COMPILER_V2=true cargo test` in e2e-move-tests or compare the result of `aptos move document` with V1 and V2

File: aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs [hunk 3]
```diff
@@ -328,10 +332,17 @@ static PRECOMPILED_APTOS_FRAMEWORK_V2: Lazy<PrecompiledFilesModules> = Lazy::new
...
+                .files()
+                .unwrap(),
+        ),
+        named_address_map: string_map_to_symbol_map(aptos_framework::named_addresses()),
+    }];
     let options = move_compiler_v2::Options {
-        sources: aptos_cached_packages::head_release_bundle()
-            .files()
-            .unwrap(),
+        packages: source_packages,
```
Comment (wrwg) [inline aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs:345]: I really would prefer we would not have v1 stuff like PackagePaths leak to the v2 code base. The current driver was intentionally build to not use this type. Why do we need it now?

File: third_party/move/tools/move-package/src/compilation/compiled_package.rs (hunk 1 above)
Comment (brmataptos) [reply third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: e2e-move-tests doesn't generate any docs.  I've seen a diff with `aptos move document` but it was a pain to set up.  We desperately need some aptos move cli tests.  (Or tests of the functionality.)

Comment (brmataptos) [reply third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: I also was able to see a difference by installing the aptos cli and running it, but that's quite a pain.  Do we have aptos move cli test cases somewhere?

Comment (rahxephon89) [reply third_party/move/tools/move-package/src/compilation/compiled_package.rs:748]: I only found the smoke test for aptos cli: https://github.com/aptos-labs/aptos-core/tree/main/testsuite/smoke-test/src/aptos_cli

**Pull request closed without merging**